tests/Unit/Ads/AssetSuggestionsServiceTest.php

```diff
@@ -324,17 +317,14 @@ public function test_get_post_assets_for_products() {
 	}
 
 	public function test_get_shop_assets() {
-		$image_post     = $this->factory()->attachment->create_upload_object( $this->get_data_file_path( 'test-image-1.png' ) );
-		$image_product  = $this->factory()->attachment->create_upload_object( $this->get_data_file_path( 'test-image-1.png' ) );
-		$this->post->ID = wc_get_page_id( 'shop' );
+		$image_post    = $this->factory()->attachment->create_upload_object( $this->get_data_file_path( 'test-image-1.png' ) );
```

Comment:
TIL that we can use factory to mock post, term and attachment, etc.

Reply:
if you like to check all of them, here is a list of factory methods :  https://github.com/bobbingwide/wordpress-develop-tests/tree/master/phpunit/includes/factory